### PR TITLE
FOUR-20025 integration with Docusign cannot be established

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps
Describe the issue this ticket solves and describe how to reproduce the issue (please attach any fixtures used to reproduce the issue).

## Solution
- Validate case number in request comments
- Validate instance case number on the case status update
- Validate instance case number on the case task status update

https://github.com/user-attachments/assets/8e60eb8c-342e-4ebd-acff-983fbe71bc1e

## How to Test
See the ticket.

## Related Tickets & Packages
[FOUR-20025](https://processmaker.atlassian.net/browse/FOUR-20025)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-20025]: https://processmaker.atlassian.net/browse/FOUR-20025?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-a13115de8a.engk8s.processmaker.net

File: tests/Feature/Cases/CaseStartedTest.php [hunk 1]
```diff
@@ -395,4 +397,26 @@ public function test_try_update_if_case_has_not_been_created()
         $repo->update($instance, $token);
         $this->assertDatabaseCount('cases_started', 1);
     }
+
+    public function test_try_store_docusign_authorization_request()
+    {
+        $category = ProcessCategoryFactory::new()->create([
```
[the remark this answers is not in the document]
Comment (devmiguelangel) [reply tests/Feature/Cases/CaseStartedTest.php:403]: The validation was added

[the remark this answers is not in the document]
Comment (caleeli) [reply tests/Feature/Cases/CaseStartedTest.php:403]: please add how many time the message should be received. e.g. `->once()`  or `->times(2)`

[the remark this answers is not in the document]
Comment (caleeli) [reply tests/Feature/Cases/CaseStartedTest.php:403]: This test an be improved by ticket: https://processmaker.atlassian.net/browse/FOUR-20093

**Review by caleeli — APPROVED** [review]

File: ProcessMaker/Repositories/TokenRepository.php [hunk 2]
```diff
@@ -690,4 +684,26 @@ public function persistEventBasedGatewayActivated(EventBasedGatewayInterface $ev
...
+     * @param TokenInterface token The `token` parameter in the `updateCaseStartedTask` function is an object of type
+     * `TokenInterface`. It is used to access the instance's case number and pass the token itself to the
+     * `CaseTaskRepository` constructor for further processing.
+     *
+     * @return void
+     */
+    private function updateCaseStartedTask(TokenInterface $token): void
+    {
+        if (is_null($token->getInstance()->case_number)) {
+            Log::error('CaseException: case number not found, method=updateCaseStartedTask, instance=' . $token->getInstance()->getKey());
```
Comment (caleeli) [inline ProcessMaker/Repositories/TokenRepository.php:700]: Consider change this log to warning

**Pull request merged**